src/Microsoft.Health.Core/Features/Health/HealthReportExtensions.cs

```diff
@@ -0,0 +1,32 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Health.Core.Features.Health;
+
+internal static class HealthReportExtensions
+{
+    public static HealthStatusReason GetHighestSeverityReason(this HealthReport healthReport, HealthStatusReason defaultReason)
+    {
+        HealthStatusReason worstReason = defaultReason;
+
+        foreach (var entry in healthReport.Entries)
+        {
+            if (entry.Value.Data.TryGetValue("Reason", out object reason))
```

Comment:
Today, our reports have 6 entries (BackgroundServiceHealthCheck, DcmHealthCheck, MetadataHealthCheck, WorkitemHealthCheck, DataStoreHealthCheck, Authorization Service), one for the result of each health check we run. 

Most of the time, `entry.Value.Data` will actually be empty. We only specify a Reason today when CMK is misconfigured, since its a specific check we make. 

We could sort the entries though - but its a bit messy since we need to convert to an Enum and check for null within the sort itself
```
        var orderedEntries = healthReport.Entries.OrderByDescending(e =>
            Enum.Parse<HealthStatusReason>(
                e.Value.Data.GetValueOrDefault("Reason")?.ToString() ??
                defaultReason.ToString()));

        return Enum.Parse<HealthStatusReason>(orderedEntries.First().Value.Data.GetValueOrDefault("Reason")?.ToString() ?? defaultReason.ToString());
```